src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -420,42 +428,37 @@ case class WdlBinding(ast: Ast, source: WdlSource, importResolver: ImportResolve
           }
 
           /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
-          val memberAccesses = WdlBinding.findAstsWithTrail(inputKv._2.ast, "MemberAccess") filter {
-            case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
-          }
-          memberAccesses.keys map validateMemberAccess
+          WdlBinding.findTopLevelMemberAccesses(inputKv._2.ast).map(partialEvalMemberAccess).map {_.get}
         }
       }
     }
   }
 
   /* Partially evaluate MemberAccess ASTs to make sure they're not nonsense at compile time */
-  def validateMemberAccess(ast: Ast): Any = {
+  def partialEvalMemberAccess(ast: Ast): Try[Call] = {
+    def callFromName(name: String): Try[Call] = {
+      val callList = workflows.head.calls.filter {_.name == name}
+      if (callList.isEmpty) {
```

Comment:
Perhaps something like (free typed, might not be correct):

    callList.headOption map {Success(_)} getOrElse Failure(...)

Or even just:

    callList.headOption match {
       ....
    }